### PR TITLE
Update/common config

### PR DESCRIPTION
**Pull request by himanshu-egov**
### [NEW]

-  Created common config for Employee and Citizen in FSM
-  Created a service for the component registry can be used as "Digit.ComponentRegistryService"
-  moved all the imported components from config to FSMModule  file

### CODE REVIEW

File: packages/modules/fsm/src/pageComponents/CheckSlum.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React, { useState } from "react";
+import { FormStep, RadioOrSelect } from "@egovernments/digit-ui-react-components";
+
+const CheckSlum = ({ t, config, onSelect, userType, formData }) => {
+  console.log("find config here", { t, config, onSelect, userType, formData });
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/CheckSlum.js:5]: remove

File: packages/modules/fsm/src/pageComponents/SelectAddress.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    const { city } = formData?.address || {};
+    return city ? city : null;
+  });
+  const [localities, setLocalities] = useState(null);
+  const [selectedLocality, setSelectedLocality] = useState(() => {
+    const { locality } = formData?.address || {};
+    return locality ? locality : null;
+  });
+
+  console.log("find address here");
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectAddress.js:28]: remove

File: packages/modules/fsm/src/pageComponents/SelectLandmark.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  function onChange(e) {
+    if (e.target.value.length > 1024) {
+      setError("CS_COMMON_LANDMARK_MAX_LENGTH");
+    } else {
+      setError(null);
+      setLandmark(e.target.value);
+      if (userType === "employee") {
+        const value = e?.target?.value;
+        const key = e?.target?.id;
+        console.log({ key, value });
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectLandmark.js:36]: remove

File: packages/modules/fsm/src/pageComponents/SelectPincode.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import { FormStep, TextInput, CardLabel, LabelFieldPair } from "@egovernments/digit-ui-react-components";
+import React, { useState, useEffect } from "react";
+
+const SelectPincode = ({ t, config, onSelect, formData = {}, userType }) => {
+  console.log("find pincode properties here", config, onSelect, formData, userType);
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectPincode.js:5]: remove

File: packages/modules/fsm/src/pageComponents/SelectPincode.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  const [pincodeServicability, setPincodeServicability] = useState(null);
+
+  useEffect(() => {
+    // console.log("find pincode datahere", formData?.address?.pincode)
+    if (formData?.address?.pincode) {
+      setPincode(formData.address.pincode);
+    }
+  }, [formData?.address?.pincode]);
+
+  console.log({ formData }, "pindocde");
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectPincode.js:33]: remove

File: packages/modules/fsm/src/pageComponents/SelectPropertyType.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from "react";
+import { CitizenInfoLabel, Loader, Dropdown, FormStep, CardLabel, RadioOrSelect } from "@egovernments/digit-ui-react-components";
+
+const SelectPropertyType = ({ config, onSelect, t, userType, formData }) => {
+  console.log("find config property here ", config);
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectPropertyType.js:5]: remove

File: packages/modules/fsm/src/pageComponents/SelectPropertyType.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    onSelect(config.key, propertyType);
+  };
+  function selectedValue(value) {
+    setPropertyType(value);
+  }
+  function selectedType(value) {
+    onSelect(config.key, value.code);
+  }
+
+  console.log("find me here");
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectPropertyType.js:37]: remove

File: packages/modules/fsm/src/pageComponents/SelectSlumName.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from "react";
+import { CardLabel, LabelFieldPair, Dropdown, FormStep } from "@egovernments/digit-ui-react-components";
+
+const SelectSlumName = ({ config, onSelect, t, userType, formData }) => {
+  console.log("find config here", { t, config, onSelect, userType, formData });
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectSlumName.js:5]: remove

File: packages/modules/fsm/src/pageComponents/SelectTankSize.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from "react";
+import { FormStep, PitDimension } from "@egovernments/digit-ui-react-components";
+
+const isConventionalSpecticTank = (tankDimension) => tankDimension === "lbd";
+
+const SelectTankSize = ({ config, onSelect, t, formData = {}, userType }) => {
+  console.log({ config, onSelect, t, formData, userType });
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pageComponents/SelectTankSize.js:7]: remove

File: packages/modules/fsm/src/pages/citizen/NewApplication/index.js [hunk 10]
```diff
@@ -15,24 +13,30 @@ const FileComplaint = ({ parentRoute }) => {
...
 
   const submitComplaint = async () => {
     history.push(`${parentRoute}/new-application/response`);
   };
 
-  function handleSelect(data) {
-    setParams({ ...params, ...data, ...{ source: "ONLINE" } });
-    goNext();
+  function handleSelect(key, data, skipStep) {
+    console.log("find saving data here", key, data, params);
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pages/citizen/NewApplication/index.js:37]: remove

File: packages/modules/fsm/src/pages/employee/EditApplication/EditForm.js [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+          setKill(true);
+          console.log("find bill slab here", billSlab.price);
+          setValue("amountPerTrip", billSlab.price);
+          setValue("amount", billSlab.price * formData.noOfTrips);
+        }
+      }
+    })();
+    // console.log("abcd2",vehicle, formData?.propertyType , formData?.subtype)
+
+    console.log("find form data here", formData);
```
Comment (saurabh-egov) [inline packages/modules/fsm/src/pages/employee/EditApplication/EditForm.js:71]: .

File: packages/react-components/src/hoc/FormComposer.js [hunk 12]
```diff
@@ -17,14 +17,14 @@ import { useTranslation } from "react-i18next";
...
   const { t } = useTranslation();
-
   const formData = watch();
 
   useEffect(() => {
     props.getFormAccessors && props.getFormAccessors({ setValue, getValues });
   }, []);
 
   function onSubmit(data) {
+    console.log({ data });
```
Comment (saurabh-egov) [inline packages/react-components/src/hoc/FormComposer.js:27]: .

File: packages/react-components/src/hoc/FormComposer.js [hunk 13]
```diff
@@ -33,10 +33,11 @@ export const FormComposer = (props) => {
   }
 
   useEffect(() => {
-    props.onFormValueChange && props.onFormValueChange(formData, formState);
+    console.log({ formData });
```
Comment (saurabh-egov) [inline packages/react-components/src/hoc/FormComposer.js:36]: .

**Review by ayan-egov — APPROVED** [review]

**Pull request merged**